python/ray/dag/compiled_dag_node.py

```diff
@@ -775,10 +773,10 @@ def _preprocess(self) -> None:
                         "the driver cannot participate in the NCCL group"
                     )
 
-            if type(task.dag_node.type_hint) == ChannelOutputType:
+            if type(dag_node.type_hint) == ChannelOutputType:
```

Comment:
what's the difference between task.dag_node vs dag_node?

Reply:
It's the same. Their is assignment earlier `dag_node = task.dag_node`. But they are used interchangeably later on. This cleans it up to always use `dag_node`.